### PR TITLE
3.6/f/decouple logwriter creation from connection setup

### PR DESCRIPTION
**Pull request by lbudai**

### CODE REVIEW

File: modules/afsocket/afsocket-dest.c [hunk 1]
```diff
@@ -312,11 +312,13 @@ afsocket_dd_setup_addresses_method(AFSocketDestDriver *self)
 }
 
 static void
-afsocket_dd_restore_connection(AFSocketDestDriver *self)
+afsocket_dd_restore_writer(AFSocketDestDriver *self)
 {
   GlobalConfig *cfg = log_pipe_get_config(&self->super.super.super);
-
-  self->writer = cfg_persist_config_fetch(cfg, afsocket_dd_format_persist_name(self, FALSE));
+  LogWriter *fetched_writer = cfg_persist_config_fetch(cfg, afsocket_dd_format_persist_name(self, FALSE));
```
Comment (juhaszviktor) [inline modules/afsocket/afsocket-dest.c]: self->writer should be NULL at this phase.
So assert that presumption, and set the self->writer directly

File: modules/afsocket/afsocket-dest.c [hunk 2]
```diff
@@ -335,6 +337,11 @@ afsocket_dd_construct_writer_method(AFSocketDestDriver *self)
 static gboolean
 afsocket_dd_setup_writer(AFSocketDestDriver *self)
 {
+  afsocket_dd_restore_writer(self);
+
+  if (self->writer && log_pipe_is_initialized((LogPipe *)self->writer))
```
Comment (juhaszviktor) [inline modules/afsocket/afsocket-dest.c]: self->writer can't be initialized at this phase, so this check is unnecessary, I think

Comment (lbudai) [inline modules/afsocket/afsocket-dest.c]: Yes, this is not needed anymore as the `log_writer_setup()` called only in the init function.
I update the PR accordingly.

**Comment by juhaszviktor** [comment]
:+1:

**Comment by ihrwein** [comment]
:+1:

**Pull request merged**